src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -102,6 +103,28 @@ public String getWsmServiceAccountToken() throws IOException {
     return creds.getAccessToken().getTokenValue();
   }
 
+  /**
+   * Obtain the user email address from an AuthenticatedUserRequest either by the easy way (directly
+   * calling getEmail()), or the harder way, calling Sam. The hard way throws a checked exception,
+   * which we want to convert to an unchecked exception here.
+   *
+   * @param userRequest - request object for this user
+   * @return - email address of user
+   */
+  public String getRequestUserEmailOrThrow(AuthenticatedUserRequest userRequest) {
+    return Optional.ofNullable(userRequest.getEmail())
+        .orElseGet(
+            () -> {
+              try {
+                return getEmailFromToken(userRequest.getRequiredToken());
+              } catch (InterruptedException e) {
+                // Treat as a 500 for now.
```

Comment:
This call is ultimately run in a flight, so I don't think we want to convert this to a checked exception. Following the pattern of other methods here, I think `getRequestUserEmailOrThrow` should throw a checked InterruptedException.
`createControlledResource` already throws this as well, so this would only be a change to `getRequestUserEmailOrThrow`.

Reply:
Yes, but not the other call site at `validateOnlySelfAssignment`. I'll convert it there.

Reply:
Good point about the call site inside `validateOnlySelfAssignment`. I think for non-Stairway calls to Sam methods that throw InterruptedException, the common pattern is to use rethrowIfSamInterrupted, e.g:
 `final String requestUserEmail = SamService.rethrowIfSamInterrupted(() -> samService.getRequestUserEmail(userRequest), "getRequestUserEmail");`
 which handles this conversion. This is all pretty new though, so I'm still getting used to that

Reply:
I don't love that pattern.